docs/src/app/(public)/blog/[...slug]/page.tsx

```diff
@@ -87,3 +88,34 @@ export async function generateStaticParams() {
     slug: slug.split('/'),
   }));
 }
+
+export async function generateMetadata(
+  { params }: BlogProps,
+  parent: ResolvingMetadata
+): Promise<Metadata> {
+  const slugPath = params.slug;
+  const slug = slugPath.join('/');
+
+  const page = await reader.collections.blog.read(slug);
+
+  const parentOGImages = (await parent).openGraph?.images || [];
+  const parentTwitterSite = (await parent).twitter?.site ?? '';
+
+  return {
+    title: page?.title,
```

Comment:
Do we need fallbacks for any of these values if `page` is `undefined`? Or a different type via a `required` sort of thing in the schema?

Reply:
Hmm I'm a bit unsure re: this.. I wonder if what they say on https://nextjs.org/docs/app/api-reference/functions/generate-metadata#default means that if `title: page?.title,` here is undefined then it will grab the `title.default` defined in the `/blog/layout.tsx`?

Reply:
It looks like if we do `title: page?.title` and that evaluates to `title: undefined` then the `<title>` is removed. But, if we conditionally add `title` then it does fall back as you're saying. I think we should change this to:

```
...(page?.title && { title: page.title })
```

What do you think?

Reply:
Riight thanks for testing that!! Will change as per your suggestion 🙌🏻

Reply:
No worries! We should examine the rest of the properties, too. And mimic the changes on the `docs` pages.

Reply:
Don't know if this way is better but I've added fallbacks in a different way now:
```
const parentTitle = (await parent).title ?? 'Blog';
const title = page?.title ?? parentTitle;
```

Looks to work nicely when page?.title is undefined, but let me know whatya think!

Reply:
Yeah, that's solid, nice work 🙌